AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -133,9 +133,9 @@ final class StorageEngine: StorageEngineBehavior {
     func onReceiveCompletion(receiveCompletion: Subscribers.Completion<DataStoreError>) {
         switch receiveCompletion {
         case .failure(let dataStoreError):
-            storageEnginePublisher.send(completion: .failure(dataStoreError))
+            log.error("RemoteSyncEngine completed with error \(dataStoreError)")
```

Comment:
Do we know if there is any other scenario where we need to handle this?

Reply:
Good question. The behavior remains unchanged even we've changed this line. 

We have updated [the way we stop the datastore plugin](https://github.com/aws-amplify/amplify-swift/pull/2863) so that it now only stops the syncEngine when the storageEngine is stopped. It is essential to maintain the subscription between the datastore plugin and the storageEngine.